### DISCUSSION TITLE
How do I specify the tokenizer in a textpredictor?

### DISCUSSION THREAD

**Discussion by Wang-Lin-boop** [autogluon/autogluon#2334] [Q&A] [answered]
For some non natural languages, they often use vocabulary and grammar that are completely different from natural languages. I wonder what will happen if I use the text backbone trained by natural languages (such as, DeBERTa, electra, or bert) in the text predictor? Can I specify a custom tokenizer when using the textpredictor?

**Comment by sxjscience** [comment, marked as answer]
You can use your own BERT model. The tutorial is here: https://auto.gluon.ai/stable/tutorials/text_prediction/customization.html#customize-hyperparameters

**Reply by Wang-Lin-boop** [reply]
Thanks, Xing-Jian. It's work!

**Comment by Wang-Lin-boop** [comment]
I put my code here maybe someone need it.
```
import sys
from transformers import AutoModel, AutoConfig, AutoTokenizer
if __name__ == '__main__':
    local_path = sys.argv[1]
    tokenizer = AutoTokenizer.from_pretrained(sys.argv[2])
    config = AutoConfig.from_pretrained(sys.argv[3])
    model = AutoModel.from_pretrained(sys.argv[3])
    model.save_pretrained(local_path)
    config.save_pretrained(local_path)
    tokenizer.save_pretrained(local_path)
```
Make sure that the two models to be fused have the same length of embedding.
